File: providers/deepseek.js
module.exports = {
  
  name: 'deepseek',
  aliases: ['ds'],
  displayName: 'DeepSeek',
  description: 'DeepSeek API — deepseek-v4 and deepseek-reasoner models',
  baseUrl: 'https://api.deepseek.com/v1',
  apiMode: 'openai_chat',
  envVars: ['DEEPSEEK_API_KEY', 'DEEPSEEK_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'deepseek-v4',
  defaultMaxTokens: 8192,
  signupUrl: 'https://platform.deepseek.com',
  supportsHealthCheck: true,
  modelProfiles: {
    'deepseek-v4': { supportsReasoning: true, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 8192 },
    'deepseek-reasoner': { supportsReasoning: true, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 8192 },
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 131072, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/deepseek.py
"""
Provider config for DeepSeek.
"""
from __future__ import annotations
INFO: dict[str, object] = {'name': 'DeepSeek', 'aliases': ['ds'], 'display_name': 'DeepSeek', 'description': 'DeepSeek API — deepseek-v4 and deepseek-reasoner models', 'base_url': 'https://api.deepseek.com/v1', 'api_mode': 'openai_chat', 'env_vars': ['DEEPSEEK_API_KEY', 'DEEPSEEK_BASE_URL'], 'auth_type': 'api_key', 'default_model': 'deepseek-v4', 'default_max_tokens': 8192, 'signup_url': 'https://platform.deepseek.com', 'supports_health_check': True, 'model_profiles': {'deepseek-v4': {'supportsReasoning': True, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 8192}, 'deepseek-reasoner': {'supportsReasoning': True, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 8192}, '*': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 131072, 'maxOutputTokens': 4096}}}

def resolveBaseUrl() -> str:
    return 'https://api.deepseek.com/v1'

def resolveApiKey(envKey: str | None=None) -> str | None:
    return envKey